homeworks/12/chat-system/src/Server.java

```diff
@@ -0,0 +1,33 @@
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class Server {
+    private ServerSocket serverSocket;
+
+    public Server(ServerSocket serverSocket) {
+        this.serverSocket = serverSocket;
+    }
+
+    public void startServer() {
+        try {
+            while (!serverSocket.isClosed()) {
+                Socket socket = serverSocket.accept();
+                System.out.println("A new client has connected!");
+                ClientHandler clientHandler = new ClientHandler(socket);
+
+                Thread thread = new Thread(clientHandler);

```

Comment:
You can use **Thread pool** so that you do not have to create new thread every time as it is more resource-intensive. [https://docs.oracle.com/javase/tutorial/essential/concurrency/pools.html](url)